### ISSUE TITLE
function to rename file?

### ISSUE THREAD

**Issue by jread-usgs** [USGS-R/sbtools#188]
This would be useful for archiving a previous file version:

filename = '2016-01-01.rds'

want to rename it to
filename = '2016-01-01_ARCHIVED.rds'

or something like that. Thoughts on why this should or should not be in sbtools?

**Comment by lawinslow** [comment]
You're talking about re-naming a file attached to an item on SB, right?

**Comment by jread-usgs** [comment]
exactly.

So instead of 
`item_rm_files(sb_id, filename)` then `item_append_files(sb_id, newfilenameoldfile)`

**Comment by aappling-usgs** [comment]
we were just looking in the GUI and don't see a way to do it there, so until/unless ScienceBase makes this feature directly available, the proposed function would just wrap the two functions @jread-usgs just mentioned. it could still be conceptually useful, though, and would clean up client code. then if ScienceBase ever offers an API version, the implementation of this sbtools function could be updated.

**Comment by lawinslow** [comment]
Yeah, I don't know of an API-direct way. One of the dangers is a failure somewhere in the middle. We'd need to Download, rename, delete old file on SB item, and then upload with new name. Most dangerous would be failure between delete and upload.

**Comment by jread-usgs** [comment]
maybe it is just best to wait on this w/ an open issue as a wontfix and ask SB what they recommend or try to push to expose a service that would do it.
---
Jordan Read, PhD | U.S. Geological Survey | Office of Water Information
608-821-3922 | water • data • science <http://cida.usgs.gov/datascience.html>

**Comment by aappling-usgs** [comment]
hmm, good point, @lawinslow. actually raises the issue of how to do this safely in mda.streams, @jread-usgs . for either/both sbtools and mda.streams, this functionality could be done in the order

download, rename, upload with new name, check for new file on SB & stop/retry if missing, delete old file on SB.

asking SB for a service is also a good idea.

**Comment by jread-usgs** [comment]
just emailed.

**Comment by jread-usgs** [comment]
response:
```
I just verified that you can change the name of file the same way you can change the title of a file using the RESTful JSON service for an Item.
We don't store files by their name, so it's just a property.
```
so maybe this can already be done with `item_update(sb_id, info, ..., session = current_session())`?

Not sure (haven't tested), but if so, should we expose the higher level `item_rename_files()`?

**Comment by lawinslow** [comment]
Yeah, this works. 

```R
it = item_get('548b2b31e4b03f64633662a4')
it$files[[2]]$name
[1] "gdp.txt"

it$files[[2]]$name = 'gdp_new.txt'
item_update(it, info = list(files=it$files))
```
![image](https://cloud.githubusercontent.com/assets/347600/13956231/e4fa810c-f015-11e5-85eb-3c359d380fe4.png)
